spec/controllers/users/totp_setup_controller_spec.rb

```diff
@@ -41,64 +41,59 @@
     context 'when user presents invalid code' do
       before do
         sign_in_as_user
+
+        stub_analytics
+        allow(@analytics).to receive(:track_event)
+
         get :new
         patch :confirm, code: 123
       end
 
-      it 'redirects back to authenticator_setup_path' do
+      it 'redirects with an error message' do
         expect(response).to redirect_to(authenticator_setup_path)
-      end
-
-      it 'sets flash[:error] message' do
         expect(flash[:error]).to eq t('errors.invalid_totp')
-      end
-
-      it 'does not enable TOTP for the current user' do
```

Comment:
I guess I meant `precision` rather than `fidelity`, sorry (http://googletesting.blogspot.com/2014/05/testing-on-toilet-effective-testing.html)

When you take a group unit tests that can each fail individually, and combine them into a single test  (that is more course grained), you loose some precision which which a test failure pinpoints what the actual failure is.  

IIUC you can simple at `:all` to get the same effect.. it will run the `:all` block just once.. then run each of the one-line unit tests, which has the exact same efficiency benefits as combining the one-line unit tests into a single test.  (I think :)